components/brave_wallet_ui/components/extension/sign-panel/sign_in_with_ethereum.tsx

```diff
@@ -0,0 +1,356 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import Button from '@brave/leo/react/button'
+import { skipToken } from '@reduxjs/toolkit/query/react'
+
+// Types
+import {
+  BraveWallet
+} from '../../../constants/types'
+
+// Queries
+import {
+  useAccountQuery
+} from '../../../common/slices/api.slice.extra'
+
+// Components
+import CreateSiteOrigin from '../../shared/create-site-origin/index'
+import {
+  CreateAccountIcon
+} from '../../shared/create-account-icon/create-account-icon'
+
+// Utils
+import {
+  reduceAddress
+} from '../../../utils/reduce-address'
+import { getLocale } from '../../../../common/locale'
+
+// Styles
+import {
+  StyledWrapper,
+  Title,
+  Description,
+  FavIcon,
+  OriginName,
+  MessageBox,
+  MessageText,
+  IconButton,
+  CloseIcon,
+  DetailsKeyText,
+  DetailsInfoText,
+  CodeBlock,
+  URLText
+} from './sign_in_with_ethereum.style'
+import {
+  Row,
+  Column,
+  VerticalDivider,
+  VerticalSpace,
+  HorizontalSpace,
+  ScrollableColumn
+} from '../../shared/style'
+
+export const getKeyLocale = (key: string) => {
+  const capitalized = key.charAt(0).toUpperCase() + key.slice(1)
+  const localeString = `braveWallet${capitalized}`
+  return getLocale(localeString) === localeString
+    ? key
+    : getLocale(localeString)
+}
+
+interface Props {
+  data: BraveWallet.SignMessageRequest
+  onSignIn: () => void
+  onCancel: () => void
+}
+
+export const SignInWithEthereum = (props: Props) => {
+  const {
+    data,
+    onSignIn,
+    onCancel
+  } = props
+
+  // State
+  const [showDetails, setShowDetails] = React.useState<boolean>(false)
+
+  // Queries
+  const { account } = useAccountQuery(
+    data.accountId || skipToken
+  )
+
+  // Methods
+  const getFormattedKeyValue =
```

Comment:
Can we refactor this so that it can be defined outside of the component?

Reply:
Moved out of Component!